calyx-py/test/correctness/queues/binheap/binheap.py

```diff
@@ -31,47 +31,46 @@ def insert_swap(prog, name, width, len, idx_w):
     return comp
 
 
-def insert_binheap(prog, name, queue_len_factor):
+def insert_binheap(prog, name, queue_size_factor):
     """Inserts the component `binheap` into the program.
 
     It is a minimum binary heap, represented as an array.
 
     It has:
     - three inputs, `cmd`, `value`, and `rank`.
-    - one memory, `mem`, of size `(2**queue_len_factor) - 1`.
+    - one memory, `mem`, of size `2**queue_size_factor`.
     - two ref registers, `ans` and `err`.
     """
 
     comp = prog.component(name)
 
-    max_queue_len = (2**queue_len_factor) - 1
+    max_queue_size = 2**queue_size_factor
+    addr_size = queue_size_factor
 
     cmd = comp.input("cmd", 2)
     # If it is 0, we pop.
     # If it is 1, we peek.
     # If it is 2, we push `(rank, value)` to the queue.
 
     # The value and associated rank to push to the heap.
-    rank = comp.input("rank", 32)
+    rank = comp.input("rank", 64)
     value = comp.input("value", 32)
 
-    swap = comp.cell(
-        "swap", insert_swap(prog, "swap", 64, max_queue_len, queue_len_factor)
-    )
-    tuplify = comp.cell("tuplify", insert_tuplify(prog, "tuplify", 32, 32))
-    untuplify = comp.cell("untuplify", insert_untuplify(prog, "untuplify", 32, 32))
+    swap = comp.cell("swap", insert_swap(prog, "swap", 96, max_queue_size, addr_size))
+    tuplify = comp.cell("tuplify", insert_tuplify(prog, "tuplify", 64, 32))
+    untuplify = comp.cell("untuplify", insert_untuplify(prog, "untuplify", 64, 32))
 
-    mem = comp.seq_mem_d1("mem", 64, max_queue_len, queue_len_factor)
+    mem = comp.seq_mem_d1("mem", 96, max_queue_size, addr_size)
     # The memory to store the heap, represented as an array.
-    # Each cell of the memory is 64 bits wide: a 32-bit rank and a 32-bit value.
+    # Each cell of the memory is 96 bits wide: a 64-bit rank and a 32-bit value.
```

Comment:
If these magic numbers are getting a bit annoying, you could also make them arguments to this Python method.

Reply:
Done in [de13dc7](https://github.com/calyxir/calyx/pull/2164/commits/de13dc7e0ea656d8e452d19f98651e75c05c13f9)